### ISSUE TITLE
Fix screen reader support for dropdown

### ISSUE THREAD

**Issue by maxatdetroit** [CityOfDetroit/COD-Design-System#243]
## The Problem

The dropdown menu does not announce opening / closing to screenreaders.

See https://github.com/CityOfDetroit/detroitmi/issues/1577 for an example of this in the wild.

## To Reproduce

1. Go to https://cityofdetroit.github.io/COD-Design-System/?path=/docs/components-molecules-dropdown--docs
2. Launch a screen reader
3. Click on the dropdown
4. Observe that the dropdown opening is not announced

**Screenshots**

![image](https://github.com/user-attachments/assets/bb6b3e9b-b1de-477e-b129-ea2453319dac)

## Expected behavior

The dropdown opening is announced.

## The Solution

This is most likely due to the lack of the following attributes on the dropdown button and menu:
- `aria-controls="[menu-id]"` on the button
- `aria-labelledby="[button-id]"` on the dropdown menu
- `aria-hidden="true/false"` on the dropdown menu

To fix this, we need to update the Dropdown.js COD design system to add and manage the state of the aforementioned attributes on the `cod-button` and `cod-dropdown-menu`. When the dropdown menu is opened/closed, the state of `aria-hidden` should be updated to reflect the state of the menu.
